In [43]:
import pandas as pd
import numpy as np

In [9]:
df_com_CIK_SIC=pd.read_csv("../../raw/stock/company_CIK_SIC.csv")

In [10]:
df_com_CIK_SIC

cik                                name entityType     sic  \
0       320193                          Apple Inc.  operating  3571.0   
1       789019                      MICROSOFT CORP  operating  7372.0   
2      1045810                         NVIDIA CORP  operating  3674.0   
3      1652044                       Alphabet Inc.  operating  7370.0   
4      1018724                      AMAZON COM INC  operating  5961.0   
...        ...                                 ...        ...     ...   
10043  1953530                   Oculis Holding AG      other  2834.0   
10044  1966734  AI Transportation Acquisition Corp  operating  6770.0   
10045  1966734  AI Transportation Acquisition Corp  operating  6770.0   
10046  1969475            Bayview Acquisition Corp  operating  6770.0   
10047  1969475            Bayview Acquisition Corp  operating  6770.0   

                                          sicDescription  \
0                                   Electronic Computers   
1                          Services-Prepackaged Software   
2                       Semiconductors & Related Devices   
3      Services-Computer Programming, Data Processing...   
4                     Retail-Catalog & Mail-Order Houses   
...                                                  ...   
10043                        Pharmaceutical Preparations   
10044                                       Blank Checks   
10045                                       Blank Checks   
10046                                       Blank Checks   
10047                                       Blank Checks   

                            ownerOrg  insiderTransactionForOwnerExists  \
0                      06 Technology                                 0   
1                      06 Technology                                 1   
2                   04 Manufacturing                                 1   
3                      06 Technology                                 1   
4                07 Trade & Services                                 1   
...                              ...                               ...   
10043               03 Life Sciences                                 0   
10044  05 Real Estate & Construction                                 0   
10045  05 Real Estate & Construction                                 0   
10046  05 Real Estate & Construction                                 0   
10047  05 Real Estate & Construction                                 0   

       insiderTransactionForIssuerExists           tickers  \
0                                      1              AAPL   
1                                      1              MSFT   
2                                      1              NVDA   
3                                      1        GOOGL,GOOG   
4                                      1              AMZN   
...                                  ...               ...   
10043                                  0         OCS,OCSAW   
10044                                  1  AITR,AITRR,AITRU   
10045                                  1  AITR,AITRR,AITRU   
10046                                  1  BAYA,BAYAR,BAYAU   
10047                                  1  BAYA,BAYAR,BAYAU   

                  exchanges  ...                    mailing_city  \
0                    Nasdaq  ...                       CUPERTINO   
1                    Nasdaq  ...                         REDMOND   
2                    Nasdaq  ...                     SANTA CLARA   
3             Nasdaq,Nasdaq  ...                   MOUNTAIN VIEW   
4                    Nasdaq  ...                         SEATTLE   
...                     ...  ...                             ...   
10043         Nasdaq,Nasdaq  ...                             ZUG   
10044  Nasdaq,Nasdaq,Nasdaq  ...  SEVEN MILE BEACH, GRAND CAYMAN   
10045  Nasdaq,Nasdaq,Nasdaq  ...  SEVEN MILE BEACH, GRAND CAYMAN   
10046  Nasdaq,Nasdaq,Nasdaq  ...                        NEW YORK   
10047  Nasdaq,Nasdaq,Nasdaq  ..

In [52]:
df_com_CIK_SIC.ownerOrg.unique()

array(['06 Technology', '04 Manufacturing', '07 Trade & Services',
       '02 Finance', '03 Life Sciences', '01 Energy & Transportation',
       nan, '08 Industrial Applications and Services',
       'International Corp Fin', '05 Real Estate & Construction',
       '09 Crypto Assets', 'Office of Structured Finance'], dtype=object)

We clearly care about "06 Technology", but we might want to take a look at "04 Manufacturing" and '08 Industrial Applications and Services' to make sure what can be considered as tech industry is not accidentally missed. 

In [53]:
df_com_CIK_SIC.loc[df_com_CIK_SIC.ownerOrg=="04 Manufacturing"].sicDescription.unique()

array(['Semiconductors & Related Devices',
       'Motor Vehicles & Passenger Car Bodies', 'Beverages',
       'Bottled & Canned Soft Drinks & Carbonated Waters',
       'Radio & Tv Broadcasting & Communications Equipment',
       'Electronic & Other Electrical Equipment (No Computer Equip)',
       'Cigarettes', 'Aircraft Engines & Engine Parts',
       'Guided Missiles & Space Vehicles & Parts',
       'Rubber & Plastics Footwear', 'Malt Beverages', 'Aircraft',
       'Household Audio & Video Equipment', 'Food and Kindred Products',
       'Ship & Boat Building & Repairing', 'Electronic Connectors',
       "Men's & Boys' Furnishgs, Work Clothg, & Allied Garments",
       'Miscellaneous Fabricated Metal Products',
       'Search, Detection, Navigation, Guidance, Aeronautical Sys',
       'Miscellaneous Publishing',
       'Aircraft Parts & Auxiliary Equipment, NEC', 'Cement, Hydraulic',
       'Converted Paper & Paperboard Prods (No Contaners/Boxes)',
       'Canned, Frozen & Preservd

In above list of industries under 04 Manufacturing, 'Semiconductors & Related Devices', 'Electronic & Other Electrical Equipment (No Computer Equip)' (excluded), 'Electronic Connectors', 'Electronic Components, NEC', 'Electronic Components & Accessories', 'Electrical Industrial Apparatus', "Games, Toys & Children's Vehicles (No Dolls & Bicycles)", 'Telephone & Telegraph Apparatus', 'Electric Lighting & Wiring Equipment', 'Miscellaneous Electrical Machinery, Equipment & Supplies', 'Communications Equipment, NEC', 'Electronic Coils, Transformers & Other Inductors', 'Power, Distribution & Specialty Transformers' can be considered to be relevant to tech. 

In [57]:
df_com_CIK_SIC.loc[df_com_CIK_SIC.ownerOrg=='08 Industrial Applications and Services'].sicDescription.unique()

array(['Soap, Detergents, Cleang Preparations, Perfumes, Cosmetics',
       'Measuring & Controlling Devices, NEC',
       'Industrial Inorganic Chemicals',
       'Industrial Instruments For Measurement, Display, and Control',
       'Orthopedic, Prosthetic & Surgical Appliances & Supplies',
       'Surgical & Medical Instruments & Apparatus',
       'Optical Instruments & Lenses',
       'Electromedical & Electrotherapeutic Apparatus',
       'Services-General Medical & Surgical Hospitals, NEC',
       'Perfumes, Cosmetics & Other Toilet Preparations',
       'Auto Controls For Regulating Residential & Comml Environments',
       'Plastic Material, Synth Resin/Rubber, Cellulos (No Glass)',
       'Ophthalmic Goods',
       'Services-Commercial Physical & Biological Research',
       'Laboratory Analytical Instruments',
       'X-Ray Apparatus & Tubes & Related Irradiation Apparatus',
       'Plastic Materials, Synth Resins & Nonvulcan Elastomers',
       'Agricultural Production-Crop

In above list of industries under 08 Industrial Applications and Services, 'Measuring & Controlling Devices, NEC', 'Industrial Instruments For Measurement, Display, and Control', 'Auto Controls For Regulating Residential & Comml Environments', 'Instruments For Meas & Testing of  Electricity & Elec Signals', 'Photographic Equipment & Supplies', 'Laboratory Apparatus & Furniture', 'Electromedical & Electrotherapeutic Apparatus' can be considered to be related to tech industry. 

In above, the "maybe related to tech" list are man made and very much subjective. We will create a list (lst_sicD_maybe_tech) to store those sicDescription's. 

In [154]:
lst_sicD_maybe_tech=['Semiconductors & Related Devices', 'Electronic Connectors', 'Electronic Components, NEC', 'Electronic Components & Accessories', 'Electrical Industrial Apparatus', "Games, Toys & Children's Vehicles (No Dolls & Bicycles)", 'Telephone & Telegraph Apparatus', 'Electric Lighting & Wiring Equipment', 'Miscellaneous Electrical Machinery, Equipment & Supplies', 'Communications Equipment, NEC', 'Electronic Coils, Transformers & Other Inductors', 'Power, Distribution & Specialty Transformers', 'Measuring & Controlling Devices, NEC', 'Industrial Instruments For Measurement, Display, and Control', 'Auto Controls For Regulating Residential & Comml Environments', 'Instruments For Meas & Testing of  Electricity & Elec Signals', 'Photographic Equipment & Supplies', 'Laboratory Apparatus & Furniture', 'Electromedical & Electrotherapeutic Apparatus']

In [155]:
lst_sicD_maybe_tech

['Semiconductors & Related Devices',
 'Electronic Connectors',
 'Electronic Components, NEC',
 'Electronic Components & Accessories',
 'Electrical Industrial Apparatus',
 "Games, Toys & Children's Vehicles (No Dolls & Bicycles)",
 'Telephone & Telegraph Apparatus',
 'Electric Lighting & Wiring Equipment',
 'Miscellaneous Electrical Machinery, Equipment & Supplies',
 'Communications Equipment, NEC',
 'Electronic Coils, Transformers & Other Inductors',
 'Power, Distribution & Specialty Transformers',
 'Measuring & Controlling Devices, NEC',
 'Industrial Instruments For Measurement, Display, and Control',
 'Auto Controls For Regulating Residential & Comml Environments',
 'Instruments For Meas & Testing of  Electricity & Elec Signals',
 'Photographic Equipment & Supplies',
 'Laboratory Apparatus & Furniture',
 'Electromedical & Electrotherapeutic Apparatus']

To clarify, what we consider to be relevant will be the following companies: 

In [156]:
df_com_CIK_SIC_rele=df_com_CIK_SIC.loc[(df_com_CIK_SIC.ownerOrg=="06 Technology") | (df_com_CIK_SIC.sicDescription.isin(lst_sicD_maybe_tech))]

In [157]:
df_com_CIK_SIC_rele

cik                                          name entityType  \
0       320193                                    Apple Inc.  operating   
1       789019                                MICROSOFT CORP  operating   
2      1045810                                   NVIDIA CORP  operating   
3      1652044                                 Alphabet Inc.  operating   
5      1326801                          Meta Platforms, Inc.  operating   
...        ...                                           ...        ...   
10002  1866226  Welsbach Technology Metals Acquisition Corp.  operating   
10012  1987240                             SCHMID Group N.V.      other   
10013  1940674          SMX (Security Matters) Public Ltd Co      other   
10019  1995194                          Heramba Electric plc      other   
10027  1965143                                Nvni Group Ltd      other   

          sic                                     sicDescription  \
0      3571.0                               Electronic Computers   
1      7372.0                      Services-Prepackaged Software   
2      3674.0                   Semiconductors & Related Devices   
3      7370.0  Services-Computer Programming, Data Processing...   
5      7370.0  Services-Computer Programming, Data Processing...   
...       ...                                                ...   
10002  3690.0  Miscellaneous Electrical Machinery, Equipment ...   
10012  3590.0  Misc Industrial & Commercial Machinery & Equip...   
10013  3590.0  Misc Industrial & Commercial Machinery & Equip...   
10019  3690.0  Miscellaneous Electrical Machinery, Equipment ...   
10027  7372.0                      Services-Prepackaged Software   

               ownerOrg  insiderTransactionForOwnerExists  \
0         06 Technology                                 0   
1         06 Technology                                 1   
2      04 Manufacturing                                 1   
3         06 Technology                                 1   
5         06 Technology                                 0   
...                 ...                               ...   
10002  04 Manufacturing                                 0   
10012     06 Technology                                 0   
10013     06 Technology                                 0   
10019  04 Manufacturing                                 0   
10027     06 Technology                                 0   

       insiderTransactionForIssuerExists           tickers  \
0                                      1              AAPL   
1                                      1              MSFT   
2                                      1              NVDA   
3                                      1        GOOGL,GOOG   
5                                      1              META   
...                                  ...               ...   
10002                                  1  WTMA,WTMAR,WTMAU   
10012                                  0        SHMD,SHMDW   
10013                                  0         SMX,SMXWW   
10019                                  0        PITA,PITAW   
10027                                  0        NVNI,NVNIW   

                  exchanges  ...   mailing_city  mailing_stateOrCountry  \
0                    Nasdaq  ...      CUPERTINO                      CA   
1                    Nasdaq  ...        REDMOND                      WA   
2                    Nasdaq  ...    SANTA CLARA                      CA   
3             Nasdaq,Nasdaq  ...  MOUNTAIN VIEW                      CA   
5                    Nasdaq  ...     MENLO PARK                      CA   
...                     ...  ...            ...                     ...   
10002  Nasdaq,Nasdaq,Nasdaq  ...        CHICAGO                      IL   
10012         Nasdaq,Nasdaq  ...   FREUDENSTADT                      2M   
10013         Nasdaq,Nasdaq  ...       DUBLIN 4                      L2   
10019         Nasdaq,Nasdaq  ...       DUBLIN 2                  

In following are the companies that was not assigned a SIC value. 

In [158]:
df_com_CIK_SIC_rele.loc[df_com_CIK_SIC.sic.isna()] # NAN values, none :)

Empty DataFrame
Columns: [cik, name, entityType, sic, sicDescription, ownerOrg, insiderTransactionForOwnerExists, insiderTransactionForIssuerExists, tickers, exchanges, ein, description, website, investorWebsite, category, fiscalYearEnd, stateOfIncorporation, stateOfIncorporationDescription, phone, flags, mailing_street1, mailing_street2, mailing_city, mailing_stateOrCountry, mailing_zipCode, mailing_stateOrCountryDescription, business_street1, business_street2, business_city, business_stateOrCountry, business_zipCode, business_stateOrCountryDescription]
Index: []

[0 rows x 32 columns]

In [159]:
df_NAICS_map_SIC=pd.read_csv("../../raw/stock/tabula-NAICS-to-SIC-Crosswalk(2017-2022).csv")

In [160]:
df_NAICS_map_SIC

SIC   NAICS                                SIC Description  \
0      116  111110                                       Soybeans   
1      119  111120                               Cash Grains, Nec   
2      119  111130                               Cash Grains, Nec   
3      111  111140                                          wheat   
4      115  111150                                           Corn   
...    ...     ...                                            ...   
2100  9631  926130        Regulation, Administration of Utilities   
2101  9641  926140           Regulation of Agricultural Marketing   
2102  9651  926150  Regulation, Miscellaneous Commercial\nSectors   
2103  9661  927110                  Space Research and Technology   
2104  9711  928110                              National Security   

                                      NAICS Description Unnamed: 4  
0                                       Soybean Farming        NaN  
1                      Oilseed (except Soybean) Farming        NaN  
2                              Dry Pea and Bean Farming        NaN  
3                                         Wheat Farming        NaN  
4                                          Corn Farming        NaN  
...                                                 ...        ...  
2100  Regulation and Administration of\nCommunicatio...        NaN  
2101  Regulation of Agricultural Marketing and\nComm...        NaN  
2102  Regulation, Licensing, and Inspection of\nMisc...        NaN  
2103                      Space Research and Technology        NaN  
2104                                  National Security        NaN  

[2105 rows x 5 columns]

It is clear that one sic might have more than one naics, for instance sic 119. We then need to find a way to assign each sic to a list (or a set) of naics's. 

In [161]:
df_NAICS_map_SIC.loc[0,"SIC"]

'116'

In [162]:
set(df_NAICS_map_SIC.loc[df_NAICS_map_SIC.SIC=="119"].NAICS)

{111120, 111130, 111150, 111191, 111199}

In [164]:
dict_SIC_NAICS={}
for ind in range(2105):
    dict_SIC_NAICS[df_NAICS_map_SIC.loc[ind,"SIC"]]=set(df_NAICS_map_SIC.loc[df_NAICS_map_SIC.SIC==df_NAICS_map_SIC.loc[ind,"SIC"]].NAICS)
    

In [165]:
dict_SIC_NAICS

{'116': {111110},
 '119': {111120, 111130, 111150, 111191, 111199},
 '111': {111140},
 '115': {111150},
 '112': {111160},
 '139': {111199, 111211, 111219, 111940, 111992, 111998, 112519},
 '134': {111211},
 '161': {111211, 111219},
 '174': {111310, 111320},
 '175': {111331, 111339},
 '172': {111332},
 '171': {111333, 111334},
 '173': {111335},
 '179': {111336, 111339},
 '182': {111411, 111419},
 '181': {111421, 111422},
 '811': {111421, 113110},
 '132': {111910},
 '131': {111920},
 '133': {111930, 111991},
 '191': {111998, 112519},
 '831': {111998, 112519, 113210},
 '919': {111998, 112512, 112519, 114112, 114119},
 '2099': {111998,
  112519,
  311212,
  311340,
  311423,
  311824,
  311830,
  311911,
  311920,
  311941,
  311942,
  311991,
  311999},
 '212': {112111},
 '241': {112111, 112120},
 '211': {112112},
 '213': {112210},
 '252': {112310},
 '251': {112320},
 '253': {112330},
 '254': {112340},
 '259': {112390},
 '214': {112410, 112420},
 '273': {112511, 112512, 112519},
 '921': {

There might be SIC values that was not assigned with a NAICS value. 

In [166]:
lst_SIC_no_NAICS=[]

for sic in df_com_CIK_SIC.sic: 
    if not np.isnan(sic):
        if str(int(sic)) not in dict_SIC_NAICS: 
            lst_SIC_no_NAICS.append(sic)

In [167]:
set(lst_SIC_no_NAICS) # a set containing all SIC that could not be assigned to a NAICS

{0.0,
 100.0,
 200.0,
 700.0,
 900.0,
 1000.0,
 1040.0,
 1090.0,
 1220.0,
 1400.0,
 1520.0,
 1540.0,
 1600.0,
 1700.0,
 2000.0,
 2020.0,
 2030.0,
 2040.0,
 2050.0,
 2060.0,
 2070.0,
 2080.0,
 2090.0,
 2100.0,
 2200.0,
 2300.0,
 2320.0,
 2330.0,
 2340.0,
 2390.0,
 2400.0,
 2430.0,
 2510.0,
 2520.0,
 2540.0,
 2600.0,
 2650.0,
 2670.0,
 2750.0,
 2780.0,
 2800.0,
 2810.0,
 2820.0,
 2833.0,
 2834.0,
 2835.0,
 2836.0,
 2840.0,
 2842.0,
 2844.0,
 2851.0,
 2860.0,
 2870.0,
 2890.0,
 2891.0,
 2990.0,
 3050.0,
 3060.0,
 3080.0,
 3081.0,
 3100.0,
 3140.0,
 3241.0,
 3260.0,
 3290.0,
 3310.0,
 3330.0,
 3350.0,
 3360.0,
 3390.0,
 3420.0,
 3430.0,
 3440.0,
 3460.0,
 3470.0,
 3480.0,
 3490.0,
 3510.0,
 3530.0,
 3540.0,
 3550.0,
 3560.0,
 3570.0,
 3576.0,
 3580.0,
 3590.0,
 3600.0,
 3620.0,
 3630.0,
 3640.0,
 3670.0,
 3678.0,
 3690.0,
 3720.0,
 3730.0,
 3760.0,
 3790.0,
 3910.0,
 3990.0,
 4210.0,
 4220.0,
 4400.0,
 4610.0,
 4700.0,
 4900.0,
 4950.0,
 4955.0,
 4991.0,
 5000.0,
 5010.0,
 5030.0,
 5040.0,

We limit our scope to relevant companies with SIC value but no NAICS value. 

In [168]:
df_com_CIK_SIC_rele

cik                                          name entityType  \
0       320193                                    Apple Inc.  operating   
1       789019                                MICROSOFT CORP  operating   
2      1045810                                   NVIDIA CORP  operating   
3      1652044                                 Alphabet Inc.  operating   
5      1326801                          Meta Platforms, Inc.  operating   
...        ...                                           ...        ...   
10002  1866226  Welsbach Technology Metals Acquisition Corp.  operating   
10012  1987240                             SCHMID Group N.V.      other   
10013  1940674          SMX (Security Matters) Public Ltd Co      other   
10019  1995194                          Heramba Electric plc      other   
10027  1965143                                Nvni Group Ltd      other   

          sic                                     sicDescription  \
0      3571.0                               Electronic Computers   
1      7372.0                      Services-Prepackaged Software   
2      3674.0                   Semiconductors & Related Devices   
3      7370.0  Services-Computer Programming, Data Processing...   
5      7370.0  Services-Computer Programming, Data Processing...   
...       ...                                                ...   
10002  3690.0  Miscellaneous Electrical Machinery, Equipment ...   
10012  3590.0  Misc Industrial & Commercial Machinery & Equip...   
10013  3590.0  Misc Industrial & Commercial Machinery & Equip...   
10019  3690.0  Miscellaneous Electrical Machinery, Equipment ...   
10027  7372.0                      Services-Prepackaged Software   

               ownerOrg  insiderTransactionForOwnerExists  \
0         06 Technology                                 0   
1         06 Technology                                 1   
2      04 Manufacturing                                 1   
3         06 Technology                                 1   
5         06 Technology                                 0   
...                 ...                               ...   
10002  04 Manufacturing                                 0   
10012     06 Technology                                 0   
10013     06 Technology                                 0   
10019  04 Manufacturing                                 0   
10027     06 Technology                                 0   

       insiderTransactionForIssuerExists           tickers  \
0                                      1              AAPL   
1                                      1              MSFT   
2                                      1              NVDA   
3                                      1        GOOGL,GOOG   
5                                      1              META   
...                                  ...               ...   
10002                                  1  WTMA,WTMAR,WTMAU   
10012                                  0        SHMD,SHMDW   
10013                                  0         SMX,SMXWW   
10019                                  0        PITA,PITAW   
10027                                  0        NVNI,NVNIW   

                  exchanges  ...   mailing_city  mailing_stateOrCountry  \
0                    Nasdaq  ...      CUPERTINO                      CA   
1                    Nasdaq  ...        REDMOND                      WA   
2                    Nasdaq  ...    SANTA CLARA                      CA   
3             Nasdaq,Nasdaq  ...  MOUNTAIN VIEW                      CA   
5                    Nasdaq  ...     MENLO PARK                      CA   
...                     ...  ...            ...                     ...   
10002  Nasdaq,Nasdaq,Nasdaq  ...        CHICAGO                      IL   
10012         Nasdaq,Nasdaq  ...   FREUDENSTADT                      2M   
10013         Nasdaq,Nasdaq  ...       DUBLIN 4                      L2   
10019         Nasdaq,Nasdaq  ...       DUBLIN 2                  

In [169]:
lst_SIC_no_NAICS_rele=[]

for sic in df_com_CIK_SIC_rele.sic: 
    if not np.isnan(sic):
        if str(int(sic)) not in dict_SIC_NAICS: 
            lst_SIC_no_NAICS_rele.append(sic)

In [170]:
set(lst_SIC_no_NAICS_rele)

{3510.0,
 3530.0,
 3540.0,
 3550.0,
 3560.0,
 3570.0,
 3576.0,
 3580.0,
 3590.0,
 3620.0,
 3640.0,
 3670.0,
 3678.0,
 3690.0,
 7370.0}

We will need to manually give them an assignment. Let's check their sicDescription first. 

In [171]:
df_com_CIK_SIC_rele.loc[df_com_CIK_SIC_rele.sic==3590.0].sample(1).sicDescription

6770    Misc Industrial & Commercial Machinery & Equip...
Name: sicDescription, dtype: object

In [172]:
for sic in set(lst_SIC_no_NAICS_rele): 
    print(sic)
    print(df_com_CIK_SIC_rele.loc[df_com_CIK_SIC_rele.sic==sic].sample(1).sicDescription)

3620.0
2088    Electrical Industrial Apparatus
Name: sicDescription, dtype: object
3550.0
731    Special Industry Machinery (No Metalworking Ma...
Name: sicDescription, dtype: object
3590.0
10013    Misc Industrial & Commercial Machinery & Equip...
Name: sicDescription, dtype: object
3560.0
626    General Industrial Machinery & Equipment
Name: sicDescription, dtype: object
7370.0
2098    Services-Computer Programming, Data Processing...
Name: sicDescription, dtype: object
3530.0
5314    Construction, Mining & Materials Handling Mach...
Name: sicDescription, dtype: object
3640.0
5803    Electric Lighting & Wiring Equipment
Name: sicDescription, dtype: object
3690.0
7009    Miscellaneous Electrical Machinery, Equipment ...
Name: sicDescription, dtype: object
3570.0
505    Computer & office Equipment
Name: sicDescription, dtype: object
3540.0
1111    Metalworkg Machinery & Equipment
Name: sicDescription, dtype: object
3510.0
8353    Engines & Turbines
Name: sicDescription, dtype: object
3

The list is short, we will just do it by hand (this will be subjective): 

3620 similar to 3629 naics {335999}

3550 similar to 3559 naics {332410, 333111, 333242, 333249, 333318}

3590 similar to 3599 naics {332710, 332813, 332999, 333318, 333999, 336390}

3560 similar to 3569 naics {314999, 333414, 333999}

7370 similar to 7374, 7371 naics {518210, 541511}

3530 similar to 3532 naics {333131}

3640 similar to 3645, 3646 naics {335121, 335122}

3690 similar to 3699 naics {333318, 333618, 333992, 335129, 335999}

3570 similar to 3571, 3572 naics {334111, 334112}

3540 similar to 3547 naics {333519}

3510 similar to 3511, 3519 naics {333611, 333618, 336390}

3670 similar to 3679 naics {333318, 334519, 339940}

3576 similar to 3571, 3572, 3575, 3578 naics {333316, 333318, 334111, 334112, 334118}

3580 similar to 3585 naics {333415}

3678 similar to 3679 naics {334220, 334310, 334418, 334419, 334515}

In [189]:
dict_SIC_NAICS["3679"]

{334220, 334310, 334418, 334419, 334515}

In [32]:
lst_com_SIC_NAICS=[dict_SIC_NAICS[str(int(sic))] for sic in df_com_CIK_SIC.sic]

KeyError: '7370'